email_verification/api/v2/views.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2019 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Email verification API
+"""
+
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import (
+    serializers,
+    status,
+)
+from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+from rest_framework.viewsets import GenericViewSet
+from rest_framework.mixins import RetrieveModelMixin
+
+from simple_email_confirmation.exceptions import EmailConfirmationExpired
+from simple_email_confirmation.models import EmailAddress
+from opencraft.swagger import (
+    VALIDATION_AND_AUTH_RESPONSES,
+    viewset_swagger_helper,
+)
+
+
+@viewset_swagger_helper(
+    retrieve="Activate a user email.",
+    public_actions=["retrieve"],
+)
+class VerifyEmailViewset(RetrieveModelMixin, GenericViewSet):
+    """
+    Public view used by the frontend to activate email addresses.
+    """
+    authentication_classes = []
+    permission_classes = [AllowAny]
+    # We don't need a queryset or serializer
+    queryset = ''
```

Comment:
@giovannicimolin, should this be set to `None` instead of `''`?

Reply:
It's not possible because the Viewset expects a queryset or a valid queryset return (and `None` isn't one of them).
Running with `None` makes Swagger fail to genenerate documentation.

This was the fastest way to properly add this view to the DRF Routers. Do you have any suggestions?

Reply:
I'm not sure why this behaviour is there. But okay to leave it as-is for the time being as I will need time to figure out if there is a better way.